job-service-db/src/main/resources/procedures/cancelJob.sql

```diff
@@ -25,11 +25,14 @@ DROP FUNCTION IF EXISTS cancel_job(
     in_job_id VARCHAR(48),
     in_short_job_id VARCHAR(48)
 );
-CREATE OR REPLACE FUNCTION cancel_job(
+DROP FUNCTION IF EXISTS cancel_job(
     in_partition_id VARCHAR(40),
     in_job_id VARCHAR(48)
+);
+CREATE OR REPLACE PROCEDURE cancel_job(
```

Comment:
Have you considered whether the change in semantics going from function to procedure has any other impact?  Is there any difference w.r.t what happens when there is an error at different stages?